File: src/0997-Find-the-Town-Judge/find_the_town_judge.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int findJudge(int n, std::vector<std::vector<int>>& trust) {
        std::vector<int> in_degree(n, 0);
        std::vector<int> out_degree(n, 0);
        std::vector<int> ans;

        for (auto t : trust) {
            out_degree[t[0] - 1] += 1;
            in_degree[t[1] - 1] += 1;
        }

        for (int i = 0; i < in_degree.size(); i++) {
            if (in_degree[i] == n - 1) {
                if (out_degree[i] == 0) {
                    ans.push_back(i + 1);
                }
            } 
        }
        if (ans.size() == 1) return ans[0];
        else -1;
    }
};

int main() {
    int n = 3;
    std::vector<std::vector<int>> trust = {{1,3}, {2,3}, {3,1}};
    
    Solution solution;

    std::cout << solution.findJudge(n, trust);
    
    return 0;
}

File: src/0997-Find-the-Town-Judge/find_the_town_judge.py
from typing import List

class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        in_degree: List[int] = [0] * n
        out_degree: List[int] = [0] * n
        ans: List[int] = []

        for t in trust:
            out_degree[t[0] - 1] += 1
            in_degree[t[1] - 1] += 1

        for i in range(len(in_degree)):
            if in_degree[i] == n - 1:
                if out_degree[i] == 0:
                    ans.append(i + 1)
        
        return ans[0] if len(ans) == 1 else -1
     

if __name__ == "__main__":
    n: int = 3
    trust: List[List[int]] = [[1,3],[2,3],[3,1]]
        
    solution = Solution()
    
    print(solution.findJudge(n, trust))